### PR TITLE
Fixes lint setup to skip embedded tinygo examples

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Signed-off-by: Adrian Cole <adrian@tetrate.io>

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -141,6 +132,7 @@ issues:
         - errcheck
         - dupl
         - gosec
+        - lll
```
Comment (codefromthecrypt) [inline .golangci.yml:135]: tests tend to have really long lines

File: pkg/binary/envoytest/util.go [hunk 2]
```diff
@@ -65,7 +70,9 @@ func Run(ctx context.Context, r binary.Runner, bootstrap string) error {
 func Kill(ctx context.Context, r binary.Runner) error {
 	r.SendSignal(syscall.SIGINT)
 	r.WaitWithContext(ctx, binary.StatusTerminated)
-	archiver.Unarchive(r.DebugStore()+".tar.gz", filepath.Dir(r.DebugStore()))
+	if err := archiver.Unarchive(r.DebugStore()+".tar.gz", filepath.Dir(r.DebugStore())); err != nil {
+		return fmt.Errorf("error killing context: %w", err)
```
Comment (codefromthecrypt) [inline pkg/binary/envoytest/util.go:74]:
```suggestion
		return fmt.Errorf("error killing runner: %w", err)
```

File: pkg/extension/workspace/toolchain/builtin/toolchain.go [hunk 3]
```diff
@@ -39,7 +39,7 @@ var (
 )
 
 // NewToolchain returns a builtin toolchain with a given configuration.
-func NewToolchain(name string, cfg *config.ToolchainConfig, workspace model.Workspace) *builtin {
+func NewToolchain(name string, cfg *config.ToolchainConfig, workspace model.Workspace) *builtin { //nolint
```
Comment (codefromthecrypt) [inline pkg/extension/workspace/toolchain/builtin/toolchain.go:42]: the lint error is that returning unexported types is confusing. I feel it probably isn't unexpected here, so disabled it.

File: .golangci.yml [hunk 4]
```diff
@@ -18,15 +18,6 @@ run:
   # timeout for analysis, e.g. 30s, 5m, default is 1m
   deadline: 5m
 
-  # which dirs to skip: they won't be analyzed;
-  # can use regexp here: generated.*, regexp is applied on full path;
-  # default value is empty list, but next dirs are always skipped independently
-  # from this option's value:
-  #   	vendor$, third_party$, testdata$, examples$, Godeps$, builtin$
-  skip-dirs:
-    - pkg/binary/envoytest
```
Comment (codefromthecrypt) [inline .golangci.yml:27]: I took out the skip-dirs thing from here as it is handled externally. we aren't vendoring anything and addressing or ignoring the lint problems in envoytest was imho a better call than the confusion of two things doing skip-dir (make and here)

**Review by mathetake — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
Added https://github.com/tetratelabs/getenvoy/issues/139 to discuss linting of templates in general

**Pull request merged**